admiral/pkg/apis/admiral/routes/handlers.go

```diff
@@ -27,8 +28,32 @@ type IdentityServiceEntry struct {
 	ClusterNames []string `json:"Clusters,omitempty"`
 }
 
+
+/*
+We expect the DNS health checker to include the query param checkifreadonly with value set to true.
+The query param is used to check if the current Admiral instance is running in Active Mode or Passive Mode (also called read only mode).
+If Running in passive  mode, the health check returns 502 which forces DNS lookup to always return reference to Admiral in Active state.
+*/
+
 func (opts *RouteOpts) ReturnSuccessGET(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(200)
+	allQueryParams:= r.URL.Query()
+	checkIfReadOnlyStringVal := allQueryParams.Get("checkifreadonly")
+	//Remove all spaces
+	checkIfReadOnlyStringVal = strings.ReplaceAll(checkIfReadOnlyStringVal," ","")
+	// checkIfReadOnlyStringVal will be empty in case ""checkifreadonly" query param is not sent in the request. checkIfReadOnlyBoolVal will be false
+	checkIfReadOnlyBoolVal, _ := strconv.ParseBool(checkIfReadOnlyStringVal)
+	if checkIfReadOnlyBoolVal{
+		admiralState := opts.RemoteRegistry.AdmiralState
+		if(*admiralState).ReadOnly{
+			//Force fail health check if Admiral is in Readonly mode
+			w.WriteHeader(502)
```

Comment:
@asushanthk , I think `502` might be ok but would you think [`503`](https://datatracker.ietf.org/doc/html/rfc7231#section-6.6.4) would be more appropriate?

Reply:
This is fixed. 503 is more appropriate